mesonbuild/modules/features/feature.py

```diff
@@ -0,0 +1,356 @@
+    disable: str
+
+    def __init__(self, state: 'ModuleState',
+                 args: List['TYPE_var'],
+                 kwargs: 'TYPE_kwargs') -> None:
+
+        super().__init__()
+
+        @typed_pos_args('features.new', str, int)
+        @typed_kwargs('features.new',
+            KwargInfo(
+                'implies',
+                (FeatureObject, ContainerTypeInfo(list, FeatureObject)),
+                default=[], listify=True
+            ),
+            KwargInfo(
+                'group', (str, ContainerTypeInfo(list, str)),
+                default=[], listify=True
+            ),
+            KwargConfilctAttr('features.new', 'detect', default=[]),
+            KwargConfilctAttr('features.new', 'args', default=[]),
+            KwargInfo('test_code', (str, File), default=''),
+            KwargInfo(
+                'extra_tests', (ContainerTypeInfo(dict, (str, File))),
+                default={}
+            ),
+            KwargInfo('disable', (str), default=''),
+        )
+        def init_attrs(state: 'ModuleState',
+                       args: Tuple[str, int],
+                       kwargs: 'FeatureKwArgs') -> None:
+            self.name = args[0]
+            self.interest = args[1]
+            self.implies = set(kwargs['implies'])
+            self.group = kwargs['group']
+            self.detect = kwargs['detect']
+            self.args = kwargs['args']
+            self.test_code = kwargs['test_code']
+            self.extra_tests = kwargs['extra_tests']
+            self.disable: str = kwargs['disable']
+            if not self.detect:
+                if self.group:
+                    self.detect = [ConflictAttr(val=f) for f in self.group]
+                else:
+                    self.detect = [ConflictAttr(val=self.name)]
+
+        init_attrs(state, args, kwargs)
+        self.methods.update({
+            'update': self.update_method,
+            'get': self.get_method,
+        })
+
+    def update_method(self, state: 'ModuleState', args: List['TYPE_var'],
+                      kwargs: 'TYPE_kwargs') -> 'FeatureObject':
+        @noPosargs
+        @typed_kwargs('features.FeatureObject.update',
+            KwargInfo('name', (NoneType, str)),
+            KwargInfo('interest', (NoneType, int)),
+            KwargInfo(
+                'implies', (
+                    NoneType, FeatureObject,
+                    ContainerTypeInfo(list, FeatureObject)
+                ),
+                listify=True
+            ),
+            KwargInfo(
+                'group', (NoneType, str, ContainerTypeInfo(list, str)),
+                listify=True
+            ),
+            KwargConfilctAttr('features.FeatureObject.update', 'detect'),
+            KwargConfilctAttr('features.FeatureObject.update', 'args'),
+            KwargInfo('test_code', (NoneType, str, File)),
+            KwargInfo(
+                'extra_tests', (
+                    NoneType, ContainerTypeInfo(dict, (str, File)))
```

Comment:
```suggestion
                'extra_tests', 
                (NoneType, ContainerTypeInfo(dict, (str, File)))),
```

These are two unrelated type arguments to KwargInfo.